src/System.IO.FileSystem/src/System/IO/UnixFileStream.cs

```diff
@@ -957,7 +957,7 @@ public override long Seek(long offset, SeekOrigin origin)
         /// <returns>The new position in the stream.</returns>
         private long SeekCore(long offset, SeekOrigin origin)
         {
-            Contract.Assert(!_fileHandle.IsClosed && _canSeek);
+            Contract.Assert(!_fileHandle.IsClosed && CanSeek);
```

Comment:
@ericstj just changed all this, but I'd guess this should be _parent.CanSeek rather than just CanSeek.  Eric?